pkg/apiserver/registry/stats/multicastgroup/rest.go

```diff
@@ -0,0 +1,127 @@
+	"antrea.io/antrea/pkg/util/k8s"
+)
+
+type REST struct {
+	statsProvider statsProvider
+}
+
+// NewREST returns a REST object that will work against API services.
+func NewREST(p statsProvider) *REST {
+	return &REST{p}
+}
+
+var (
+	_ rest.Storage = &REST{}
+	_ rest.Scoper  = &REST{}
+	_ rest.Getter  = &REST{}
+	_ rest.Lister  = &REST{}
+)
+
+type statsProvider interface {
+	ListMulticastGroups() []statsv1alpha1.MulticastGroup
+	GetMulticastGroup(name string) (*statsv1alpha1.MulticastGroup, bool)
+}
+
+func (r *REST) New() runtime.Object {
+	return &statsv1alpha1.MulticastGroup{}
+}
+
+func (r *REST) NewList() runtime.Object {
+	return &statsv1alpha1.MulticastGroupList{}
+}
+
+func (r *REST) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
+	if !features.DefaultFeatureGate.Enabled(features.Multicast) {
+		return &statsv1alpha1.MulticastGroup{}, nil
+	}
+	labelSelector := labels.Everything()
+	if options != nil && options.LabelSelector != nil {
+		labelSelector = options.LabelSelector
+	}
+	members := r.statsProvider.ListMulticastGroups()
+	items := make([]statsv1alpha1.MulticastGroup, 0, len(members))
+	for i := range members {
+		if labelSelector.Matches(labels.Set(members[i].Labels)) {
+			items = append(items, members[i])
+		}
+	}
+	metricList := &statsv1alpha1.MulticastGroupList{
+		Items: items,
+	}
+	return metricList, nil
+}
+
+func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	if !features.DefaultFeatureGate.Enabled(features.Multicast) {
+		return &statsv1alpha1.MulticastGroupList{}, nil
+	}
+	metric, exists := r.statsProvider.GetMulticastGroup(name)
+	if !exists {
+		return nil, errors.NewNotFound(statsv1alpha1.Resource("multicastgroups"), name)
+	}
+	return metric, nil
+}
+
+func (r *REST) ConvertToTable(ctx context.Context, obj runtime.Object, tableOptions runtime.Object) (*metav1.Table, error) {
+	table := &metav1.Table{
+		ColumnDefinitions: []metav1.TableColumnDefinition{
+			{Name: "Group", Type: "string", Format: "name", Description: "IP of multicast group."},
+			{Name: "Pods", Type: "string", Description: "List of Pods the has joined the multicast group."},
+		},
+	}
+	if m, err := meta.ListAccessor(obj); err == nil {
+		table.ResourceVersion = m.GetResourceVersion()
+		table.Continue = m.GetContinue()
+		table.RemainingItemCount = m.GetRemainingItemCount()
+	} else {
+		if m, err := meta.CommonAccessor(obj); err == nil {
+			table.ResourceVersion = m.GetResourceVersion()
+		}
+	}
+
+	var err error
+	table.Rows, err = metatable.MetaToTableRow(obj, func(obj runtime.Object, m metav1.Object, name, age string) ([]interface{}, error) {
+		stats := obj.(*statsv1alpha1.MulticastGroup)
+		podNameSlice := make([]string, len(stats.Pods))
+		for i, pod := range stats.Pods {
+			podNameSlice[i] = k8s.NamespacedName(pod.Namespace, pod.Name)
+		}
+		return []interface{}{stats.Group, strings.Join(podNameSlice, ", ")}, nil
```

Comment:
Will this be output regardless of its length or kubectl will cut it?

Reply:
test with an extremely long string.It seems kubectl won't cut it

Reply:
then we should cut it and append a number and ... like other commands since there will be likely some Pods in a Group?

Reply:
updated. Cannot find public func in related libraries, find one in kubernetes repo https://github.com/kubernetes/kubernetes/blob/bb7b0b6a91e9f365025d496cb267817e777c872b/pkg/printers/internalversion/printers.go#L667